umbral/utils.py

```diff
@@ -55,6 +55,42 @@ def hash_to_bn(list, params):
  
     return res
 
+def hash_to_point(curve, data, constant=None):
+    """
+    Hashes arbitrary data into a valid EC point of the specified curve,
+    using the try-and-increment method.
+    It admits an optional constant as an additional input to the hash function.
+    It uses SHA256 as the internal hash function. 
+
+    WARNING: Do not use when the input data is secret, as this implementation is not 
+    in constant time, and hence, it is not safe with respect to timing attacks.
+
+    TODO: Check how to uniformly generate ycoords. Currently, it only outputs points 
```

Comment:
Probably could do this by generating the noncompressed point via SHA512, actually.

Reply:
Actually, yeah, I think this would work very well. @cygnusv What're your thoughts on this?

Reply:
But then the ycoord is not related to the xcoord. I don't follow you

Reply:
Oh, right. I forgot about it being a Galois field. I'll think more on this.

Reply:
The method is to for a given xcoord, compute the ycoord according to the curve equation and check if its a quadratic residue. I didn't want to do that, so I'm just using the fact that OpenSSL deserialization of compressed points checks everything implicitly, and if it fails, then some exception is raised.